src/main/java/com/google/step/model/User.java

```diff
@@ -1,20 +1,28 @@
 package com.google.step.model;
 
+import java.util.Optional;
+
+/** A class representing a user. */
 public class User {
   public final long id;
-  public final String name;
   public final String username;
   public final String email;
-  public final String photoBlobkey;
+  public final Optional<String> photoBlobKey;
   public final String bio;
 
-  public User(
-      long id, String name, String username, String email, String photoBlobkey, String bio) {
+  public User(long id, String email) {
+    this.id = id;
+    this.username = email;
+    this.email = email;
+    this.photoBlobKey = Optional.empty();
+    this.bio = "";
+  }
+
+  public User(long id, String email, String username, Optional<String> photoBlobKey, String bio) {
```

Comment:
Having an `Optional` as an argument is kind of weird. It is discouraged. Rather provide two separate constructors, one that takes `photoBlobKey` as argument and one that just sets it to `Optional.empty()`.
To avoid repetition of code you can also call one constructor from another.

Reply:
I see. but I cannot call the other constructor because the field is final.